### PR TITLE
case table CollectionTitle improvements

### PR DESCRIPTION
**Pull request by hmorgancode**

### CODE REVIEW

**Comment by hmorgancode** [comment]
ah, codecov has informed me that I did not add coverage for this

**Review by kswenson — CHANGES_REQUESTED** [review]
👍 This is heading in the right direction. I had two broad suggested improvements, one of which I ended up implementing myself because I wanted to test it out before suggesting it:
1. Re-simplify the props of the `CollectionTitle` component for reasons given in inline comments. (This one I left for you.)
2. Simplify the `CollectionTitle` implementation so it's not relying on render counts. (This one I ended up implementing in an additional commit. See what you think and we can discuss tomorrow if you'd like.)

File: v3/src/components/case-table/collection-title.tsx [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+  // React-data-grid takes several renders to arrive at its full width,
+  // even though the mobx data relevant to the title are available before that.
+  // So, we hide the title until the data table is fully rendered, and then
+  // calculate the correct size. This is preferable to a setTimeout,
+  // but if there's a best-practice mobX solution, I'm open to it.
+  useEffect(() => {
+    if (caseTableRenderCount < 5) {
+      setCaseTableRenderCount(caseTableRenderCount + 1)
+      return
+    }
```
[the remark this answers is not in the document]
Comment (hmorgancode) [reply v3/src/components/case-table/collection-title.tsx]: Good call! I was unaware of ResizeObservers

File: v3/src/components/case-table/collection-title.tsx [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+  const handleChangeTitle = (nextValue?: string) => {
+    if (nextValue) {
+      setTitle(nextValue)
+    }
+  }
+
+  return (
+    <div className="collection-title-wrapper" ref={titleRef}>
+      <div className="collection-title" style={titleStyle}>
+        <Editable value={isEditing ? title : `${title} (${caseCount} cases)`}
```
Comment (kswenson) [inline v3/src/components/case-table/collection-title.tsx:73]: Replace `cases` with an appropriate translated string. Something like the following from `SummaryView`:
```typescript
  const casesStr = t(caseCount === 1 ? "DG.DataContext.singleCaseName" : "DG.DataContext.pluralCaseName")
```

**Review by kswenson — APPROVED** [review]
👍 Looks good -- just one remaining issue of a hard-coded string

**Pull request merged**